iodata/wfx.py

```diff
@@ -0,0 +1,290 @@
+                re.findall(r'<MO Number>\n(.*?)\n</MO Number>', mo, re.S)]
+    # raw-primitive expansion coefficients for MO
+    coefficient_all = re.findall(
+        r'[-+]?\d+.\d+[E,e][+-]\d+', mo, flags=re.MULTILINE)
+    mo_coefficients = np.array(
+        coefficient_all, dtype=np.float).reshape(-1, num_primitives)
+    mo_coefficients = np.transpose(mo_coefficients)
+    return mo_count, mo_coefficients
+
+
+def _check_tag(f_content: str):
+    tags_header = re.findall(r'<(?!/)(.*?)>', f_content)
+    tags_tail = re.findall(r'</(.*?)>', f_content)
+    # Check if header or tail tags match
+    # head and tail tags of Molecular Orbital Primitive Coefficients are
+    # not matched paired because there are MO between
+    assert ('Molecular Orbital Primitive Coefficients' in tags_header) and \
+           ('Molecular Orbital Primitive Coefficients' in tags_tail), \
+        "Molecular Orbital Primitive Coefficients tags are not shown in " \
+        "WFX inputfile pairwise or both are missing."
+    # check others
+    tags_header_check = [i for i in tags_header
+                         if i != 'Molecular Orbital Primitive Coefficients']
+    tags_tail_check = [i for i in tags_tail
+                       if i != 'Molecular Orbital Primitive Coefficients']
+    for tag_header, tag_tail in zip(tags_header_check, tags_tail_check):
+        assert (tag_header == tag_tail), \
+            "Tag header %s and tail %s do not match." \
+            % (tag_header, tag_tail)
+    # Check if all required tags/fields are present
+    tags_required = ['Title',
+                     'Keywords',
+                     'Number of Nuclei',
+                     'Number of Primitives',
+                     'Number of Occupied Molecular Orbitals',
+                     'Number of Perturbations',
+                     'Nuclear Names',
+                     'Nuclear Charges',
+                     'Nuclear Cartesian Coordinates',
+                     'Net Charge',
+                     'Number of Electrons',
+                     'Number of Alpha Electrons',
+                     'Number of Beta Electrons',
+                     'Primitive Centers',
+                     'Primitive Types',
+                     'Primitive Exponents',
+                     'Molecular Orbital Occupation Numbers',
+                     'Molecular Orbital Energies',
+                     'Molecular Orbital Spin Types',
+                     'Molecular Orbital Primitive Coefficients',
+                     'MO Number',
+                     'Energy = T + Vne + Vee + Vnn',
+                     'Virial Ratio (-V/T)']
+    if set(tags_header).intersection(set(tags_required)) != \
+            set(tags_required):
+        diff = set(tags_required) - set(tags_header).intersection(
+            set(tags_required))
+        err_str = ', '.join(diff)
+        err_str += 'are/is required but not present in the WFX file.'
+        raise AssertionError(err_str)
```

Comment:
`AssertionError` is typically only used for logical consistency checks inside a code, normally something a user of a library should encounter. `ValueError` or `IOError` would be more suitable.

Reply:
Going to fix it soon. Thank you for the update.

Reply:
Fixed at 77dca55